apollo-common/src/main/java/com/ctrip/framework/apollo/common/entity/BaseEntity.java

```diff
@@ -118,6 +115,16 @@ public void setDataChangeLastModifiedTime(Date dataChangeLastModifiedTime) {
     this.dataChangeLastModifiedTime = dataChangeLastModifiedTime;
   }
 
+  private void checkDeletedAt() {
```

Comment:
I think your proposal is more clear.
BTW, do you think we need to add the check condition to handle cases like the entity is set deleted to true multiple times?
```java
    this.isDeleted = deleted;
    if (this.isDeleted && this.deletedAt == 0) {
      this.deletedAt = System.currentTimeMillis();
    } else if (!this.isDeleted && this.deletedAt != 0) {
      this.deletedAt = 0;
    }
```

Reply:
To avoid setting`deletedAt` multiple times, I think it is good. I prefer to do it like this:

```java
public void setDeleted(boolean deleted) {
    isDeleted = deleted;
    if (deleted && this.deletedAt == 0) {
        // also set deletedAt value as epoch millisecond
        this.deletedAt = System.currentTimeMillis();
    } else {
        this.deletedAt = 0L:
    }
}
```

Meanwhile, we should remove the setter method of `deletedAt`,  This field does only set the value when the`isDeleted` field was updated, the getter still keeps on. And currently, there are no usage references with`BaseEntity#setDeletedAt`.

Reply:
There is a bug in the following code that if `deleted == true` and `this.deletedAt != 0`, then `this.deletedAt` will be set to `0L`.
```java
public void setDeleted(boolean deleted) {
    isDeleted = deleted;
    if (deleted && this.deletedAt == 0) {
        // also set deletedAt value as epoch millisecond
        this.deletedAt = System.currentTimeMillis();
    } else {
        this.deletedAt = 0L:
    }
}
```

The updated logic is as follows:

```java
  public void setDeleted(boolean deleted) {
    isDeleted = deleted;
    if (deleted && this.deletedAt == 0) {
      // also set deletedAt value as epoch millisecond
      this.deletedAt = System.currentTimeMillis();
    } else if (!deleted) {
      this.deletedAt = 0L;
    }
  }
```

Reply:
LGTM